packages/core/src/interpreter.ts

```diff
@@ -429,12 +429,16 @@ export class Interpreter<
     this.listeners.add(listener);
 
     // Send current state to listener
-    if (this.status === InterpreterStatus.Running) {
+    if (this.status !== InterpreterStatus.NotStarted) {
       listener(this.state);
     }
 
     if (resolvedCompleteListener) {
-      this.onDone(resolvedCompleteListener);
+      if (this.status === InterpreterStatus.Stopped) {
+        resolvedCompleteListener();
```

Comment:
can't call this within `onDone` easily because there the callback expects to receive a `doneInvoke` event. Here we know that it's a parameter-less `observer.complete` though.